src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -85,66 +106,48 @@ internal partial class PackageInstallerService : AbstractDelayStartedService, IP
                               SymbolSearchOptions.SuggestForTypesInNuGetPackages)
         {
             _workspace = workspace;
+
             _serviceProvider = serviceProvider;
+            // MEFv2 doesn't support type based contract for Import above and for this particular contract
+            // (SAsyncServiceProvider) actual type cast doesn't work. (https://github.com/microsoft/vs-mef/issues/138)
+            // workaround by getting the service as object and cast to actual interface
+            _asyncServiceProvider = (Shell.IAsyncServiceProvider)asyncServiceProvider;
+
             _editorAdaptersFactoryService = editorAdaptersFactoryService;
             _packageInstallerServices = packageInstallerServices;
             _packageInstaller = packageInstaller;
             _packageUninstaller = packageUninstaller;
             _packageSourceProvider = packageSourceProvider;
         }
 
-        public async ValueTask<ImmutableArray<PackageSource>?> TryGetPackageSourcesAsync(bool allowSwitchToMainThread, CancellationToken cancellationToken)
+        public event EventHandler? PackageSourcesChanged;
+
+        public ImmutableArray<PackageSource> TryGetPackageSources()
         {
-            // Only read from _packageSourcesAsync once, since OnSourceProviderSourcesChanged could reset it to default
-            // at any time while this method is running.
-            JoinableTask<ImmutableArray<PackageSource>?> packageSourcesAsync;
+            Task<ImmutableArray<PackageSource>> localPackageSourcesTask;
             lock (_gate)
             {
-                if (_packageSourcesAsync is null)
-                {
-                    _packageSourcesAsync = ThreadingContext.JoinableTaskFactory.RunAsync(() => GetPackageSourcesImplAsync());
-                }
+                if (_packageSourcesTask is null)
+                    _packageSourcesTask = Task.Run(() => GetPackageSourcesAsync(), this.DisposalToken);
```

Comment:
```suggestion
                _packageSourcesTask ??= Task.Run(GetPackageSourcesAsync, this.DisposalToken);
```

Reply:
i'm ok as is.  i tend to like explicit lambdas in cases like this. :)